webapp/src/pages/boardPage/boardPage.tsx

```diff
@@ -229,28 +257,62 @@ const BoardPage = (props: Props): JSX.Element => {
     }, [])
 
     useEffect(() => {
-        dispatch(loadAction(match.params.boardId))
+        if (!match.params.boardId) {
+            return
+        }
+        dispatch(setCurrentBoard(match.params.boardId))
 
-        if (match.params.boardId) {
-            // set the active board
-            dispatch(setCurrentBoard(match.params.boardId))
-
-            if (viewId !== Constants.globalTeamId) {
-                // reset current, even if empty string
-                dispatch(setCurrentView(viewId))
-                if (viewId) {
-                    // don't reset per board if empty string
-                    UserSettings.setLastViewId(match.params.boardId, viewId)
-                }
+        if (viewId !== Constants.globalTeamId) {
+            dispatch(setCurrentView(viewId))
+            if (viewId) {
+                UserSettings.setLastViewId(match.params.boardId, viewId)
             }
         }
-    }, [teamId, match.params.boardId, viewId, me?.id])
 
+        dispatch(loadAction(match.params.boardId))
+    }, [teamId, match.params.boardId, viewId, me?.id, dispatch, loadAction])
+
+
+    useEffect(() => {
```

Comment:
This creates a new bug. If you have two teams with following boards-

1. Team A - Board A, B and C
2. Team B - Board D, E, F

If you open team B in boards and open board E, navigate back to Channels, then back to boards, it should stay you on Board E, but currently it navigates to first board D.

Additionally, if you open Team B Board E, then switch to team A, then to team B, it ends up switching to board D but it should stay on the board which was previously open.

I'm not sure whats the need of resetting board ID on team switch when we store each team's last board ID in local storage.